File: frontend/src/services/aiService.js
import apiClient from './apiClient';

const aiService = {
  async checkSymptoms(payload) {
    const { data } = await apiClient.post('/ai/symptom-check', payload);
    return data.data;
  },
  async summarizeReportText(payload) {
    const { data } = await apiClient.post('/ai/reports/summarize-text', payload);
    return data.data;
  },
  async summarizeReportPdf(file, labReportId) {
    const form = new FormData();
    form.append('file', file);
    if (labReportId) form.append('labReportId', labReportId);
    const { data } = await apiClient.post('/ai/reports/summarize-pdf', form, {
      headers: { 'Content-Type': 'multipart/form-data' },
    });
    return data.data;
  },
  async explainPrescription(prescriptionId) {
    const { data } = await apiClient.get(`/ai/prescriptions/${prescriptionId}/explain`);
    return data.data;
  },
  async chat(payload) {
    const { data } = await apiClient.post('/ai/chat', payload);
    return data.data;
  },
};

export default aiService;

File: backend/src/main/java/com/hospital/hms/service/AiService.java
package com.hospital.hms.service;

import com.hospital.hms.dto.request.ChatRequest;
import com.hospital.hms.dto.request.SymptomCheckRequest;
import com.hospital.hms.dto.response.ChatResponse;
import com.hospital.hms.dto.response.PrescriptionExplanationResponse;
import com.hospital.hms.dto.response.ReportSummaryResponse;
import com.hospital.hms.dto.response.SymptomCheckResponse;
import org.springframework.web.multipart.MultipartFile;

public interface AiService {

    SymptomCheckResponse checkSymptoms(SymptomCheckRequest request);

    /** Summarizes raw report text; if labReportId is present, saves the summary onto that report. */
    ReportSummaryResponse summarizeReportText(String reportText, Long labReportId);

    /** Extracts text from an uploaded PDF, then summarizes it the same way. */
    ReportSummaryResponse summarizeReportPdf(MultipartFile file, Long labReportId);

    PrescriptionExplanationResponse explainPrescription(Long prescriptionId, Long requestingUserId);

    ChatResponse chat(ChatRequest request);
}
